### PR TITLE
Proper error values

### PR DESCRIPTION
**Pull request by grouma**
Allow configurable exit codes.

Towards https://github.com/dart-lang/build/issues/1283

We still need to provide better messaging for merge errors.

### CODE REVIEW

File: build_runner/lib/src/generate/build_result.dart [hunk 1]
```diff
@@ -56,6 +64,16 @@ enum BuildStatus {
   failure,
 }
 
+/// The type of failure
+///
+/// Values correspond to proposed exitCode.
+class FailureType {
+  static const general = const FailureType(1);
+  static const cantCreate = const FailureType(73);
+  final int value;
```
Comment (jakemac53) [inline build_runner/lib/src/generate/build_result.dart]: nit: rename exitCode?

Comment (natebosch) [reply build_runner/lib/src/generate/build_result.dart]: +1

File: build_runner/lib/src/generate/build_result.dart [hunk 2]
```diff
@@ -56,6 +64,16 @@ enum BuildStatus {
...
+  const FailureType(this.value);
```
Comment (kevmoo) [inline build_runner/lib/src/generate/build_result.dart]: Make this private? Keep the set of exit codes restricted to the ones provided here?

File: build_runner/lib/src/generate/build_result.dart (hunk 1 above)
Comment (grouma) [reply build_runner/lib/src/generate/build_result.dart]: package:io doesn't expose those values but I have updated the name to exitCode.

**Pull request merged**